esp-hal-common/src/embassy.rs

```diff
@@ -0,0 +1,97 @@
+use core::{cell::Cell, ptr};
+
+use embassy_time::driver::{AlarmHandle, Driver};
+
+use crate::clock::Clocks;
+
+#[cfg_attr(
+    all(
+        feature = "embassy-time-systick",
+        any(
+            feature = "esp32c3",
+            feature = "esp32s3",
+            feature = "esp32s2",
+            feature = "esp32c2"
+        )
+    ),
+    path = "embassy/time_driver_systimer.rs"
+)]
+#[cfg_attr(
+    all(
+        feature = "embassy-time-timg",
+        any(feature = "esp32", feature = "esp32s3", feature = "esp32s2")
```

Comment:
Because the timg on the c3 (and I think c2) only has one configurable alarm, so the systick should be used instead.

Reply:
Okay, that's fine for now then. I do think we either need to document this or define a new configuration symbol in our build script just because it's not immediately clear what's going on by looking at the code. But this can be done later, won't block the PR on it.